src/v/security/scram_algorithm.cc

```diff
@@ -88,6 +88,14 @@ struct server_final_match {
     bytes signature;
 };
 
+/*
```

Comment:
Why do we need to support older versions, I would have thought we pin it?

Reply:
I was trying to minimize the number of dependencies built from source for the open-source build. This operator cast workaround was all that was needed to be compatible with the version of re2 that ships with fedora.